### ISSUE TITLE
Incorrect input validation on confirm delete

### ISSUE THREAD

**Issue by wdhdev** [dubinc/dub#1324]
If you create a short link with `*` in the slug, when attempting to delete it, the input validation says it is not valid:

![image](https://github.com/user-attachments/assets/e1b23b46-c4c1-43fd-a9fb-4dc4845ceb6b)

Browser: Firefox 131

**Comment by sudipb7** [comment]
I tested it. If you create a short link with * in the slug, for example, `hrsn.link/*` you'll have to type `hrsn.link` to validate the input.

**Comment by wdhdev** [comment]
Yep, that works, however it is still technically a validation issue that probably should be fixed. Either by disallowing the `*` character when creating links, or by just somehow fixing the frontend validation for deleting a link with the `*` character in it.

**Comment by sudipb7** [comment]
I guess both are needed as there can be existing links with `*` and user should not be able to create links like this

**Comment by ShashankGupta10** [comment]
Can i work on this @steven-tey ?
